### PR TITLE
Download translations in bulk to prevent GlotPress throttling

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #2184 by @dhruvkb

## Description

This PR adds bulk downloading for translations by logging into GlotPress, downloading the ZIP and extracting the JSON content. It reuses the existing framework for massaging them from `jed1x` to Vue i18n's format and then writing them to the destination dir.

If GlotPress credentials are not present in the environment, the old throttled approach will be used as the fallback, a Make WordPress membership is not needed for development.

## Other improvements

- Using the bulk download we get the translation strings for Latin (`la.json`) too. Unfortuanately we can't use it because it's not in the valid locales list.
- This PR also removes the redundant file `locales-list.json` that was not being used anywhere and had no references pointing to it.

## Testing Instructions

0. Check out the PR.
1. Run the following command:
   ```
   $ pnpm i18n
   ```
   You will see the parallelised download that might throttle.
2. Clear out the translations files.
3. Run the following command: 
   ```
   $ env GLOTPRESS_USERNAME="<WP.org username>" GLOTPRESS_PASSWORD="<WP.org password>" pnpm i18n
   ```
   You will see the bulk download that does not throttle.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
I just noticed that using the bulk download, we're getting another locale that we weren't getting before: `la.json` for Latin!

**Comment by dhruvkb** [comment]
**Success update:**

With the right secrets in the actions, all 200 translations are being downloaded...

Screenshot 2023-02-15 at 6 45 43 PM

...and incorporated into the Docker image.

Screenshot 2023-02-15 at 6 45 10 PM

File: src/locales/scripts/get-translations.js [hunk 1]
```diff
@@ -144,11 +241,28 @@ if (process.argv.includes("--watch")) {
...
-      console.log(`Successfully saved ${res.length + 1} translations.`)
-    })
-    .catch(console.error)
+    getAuthCookies(username, password)
+      .then(fetchBulkJed1xTranslations)
+      .then(extractZip)
+      .then((res) => {
+        console.log(`Successfully saved ${res.length + 1} translations.`)
+      })
+      .catch(console.error)
```
Comment (zackkrida) [inline src/locales/scripts/get-translations.js:255]:
```suggestion
      .catch(console.error)
```

This is a bit verbose, as it logs out the entire Axios result. A custom message withg the status code might be more legible. 

Should we also do the parallel download if the bulk fails instead?

**Review by zackkrida — APPROVED** [review]
I didn't test Docker, but this works perfectly. I left a non-blocking suggestion for the unlikely use case where invalid w.org credentials are supplied.

**Comment by dhruvkb** [comment]
@zackkrida I'm very bad with JS Promises, even getting this PR to this usable state was very difficult for me. If you can push those changes for improved logging or fallback, that'll be very helpful.

**Comment by dhruvkb** [comment]
@zackkrida my grasp on Promises was holding me back, I replaced all of that with `async`/`await` in #2192. Will merge that into this PR if you agree with those changes.

**Review by zackkrida — APPROVED** [review]
LGTM

**Review by obulat — APPROVED** [review]
LGTM! Thank you for fixing these problems, @dhruvkb!

**Pull request merged**